client/src/backoffice/EventsPage.tsx

```diff
@@ -59,55 +49,102 @@ const EditEventForm: React.FC<
   return <EventForm event={event} />;
 };
 
+type EventEditValues = {
+  name: string;
+  year: number;
+  id: number;
+  description: string;
+  start_date: string;
+  end_date: string;
+  city: string;
+  country: string;
+  event_url: string;
+  image: Blob;
+  isFile: boolean;
+};
+
 const EventForm: React.FC<{ create?: boolean; event?: PoapEvent }> = ({ create, event }) => {
-  const values = useMemo(() => {
+  const startingDate = new Date('1 Jan 1900');
+  const endingDate = new Date('1 Jan 2200');
+  const dateFormatter = (day: Date | number) => format(day, 'MM-dd-yyyy');
+
+  const { addToast } = useToasts();
+
+  const initialValues = useMemo(() => {
     if (event) {
       return {
         ...event,
+        isFile: false,
       };
     } else {
       const now = new Date();
       const year = now.getFullYear();
-      const day = now.getDate() < 10 ? `0${now.getDate()}` : now.getDate().toString();
-      const month = now.getMonth() < 10 ? `0${now.getMonth()}` : now.getMonth().toString();
-      return {
+      const values: EventEditValues = {
         name: '',
-        year: now.getFullYear(),
+        year,
         id: 0,
-        fancy_id: '',
         description: '',
-        start_date: `${year}-${month}-${day}`,
-        end_date: `${year}-${month}-${day}`,
+        start_date: '',
+        end_date: '',
         city: '',
         country: '',
         event_url: '',
-        image_url: '',
-        signer_ip: '',
-        signer: '',
+        image: new Blob(),
+        isFile: true,
       };
+      return values;
     }
   }, [event]);
+
+  const handleFileChange = (e: React.ChangeEvent<HTMLInputElement>, setFieldValue: Function) => {
+    e.preventDefault();
+    const file = e.target.files && e.target.files[0];
+
+    setFieldValue('image', file);
+  };
```

Comment:
Prefer using descriptive names instead of shorthands, prefer object/array destructuring and early returns.

Code example
```typescript
  const handleFileChange = (event: React.ChangeEvent<HTMLInputElement>, setFieldValue: () => void) => {
    event.preventDefault();
    const { files } = event.target;
    if (!files) return;

    const [firstFile] = files;
    setFieldValue('image', firstFile);
  };
```

Reply:
great. I need a little more time to get used to array destructuring though, but I like it. Same thing with early returns